### ISSUE TITLE
Emoji function.

### ISSUE THREAD

**Issue by mansibang** [klausw/hackerskeyboard#765]
It needs a emoji function so we can

**Comment by pthfdr-42** [comment]
I object. Please never do it.
From Reddit:
>I think that both emojis and even acronyms would count as Newspeak. Both types of communication we use everyday through technology (we even are starting to use some acronyms in our speech). We use emojis to make communication through text simpler and more visual. They often generate different feelings and thoughts to different people. This makes the meaning slightly different for most people. The way emojis would act like Newspeak is you can't use them to argue anything. You might be able to make a statement using one or multiple emojis, but never actually make an argument. This was the entire intention of Newspeak. It was invented so that eventually everyone in Oceania would speak it. It would take the impossibility of doubting, hating or arguing against Big Brother to a whole new level. This new level would be to the point where someone couldn't even think of anything argumentative, only statements, making any sort of thought against Big Brother or the party impossible.

**Comment by LinuxRunsTheWorld** [comment]
It takes a millisecond to switch to gboard, pick your emoji, and another millisecond to switch back to Hacker's. 

That hasn't stopped me from using it daily since its first release!

**Comment by T0MuX** [comment]
I disagree. Everybody (or, a big lot of people atleast) uses emojis everydays. I really enjoy them because I think they are really useful, making some communications really shorter and efficient.
So, what about the people part also enjoying Hacker's Keyboard ? Like me ? It's a really good project, I love it and use it everyday since 2 or 3 years ago. But I also use emojis everydays. And even it is not so long to switch to another keyboard, select emojis, and switch back to HB, it is a really pain in the ass to do it quite each time I send a text message on any communication app (riot, social network, sms...).
To me, it's just a non sense to not implemente a little and very useful tiny button with all emojis. I respect your choice, but we are 2019 ! No offense. It is just a point of view in the middle of a lot of others ones. Sorry for my approximative english. Have a nice day

**Comment by trenttobler** [comment]
If adding emoji is staightforward, it would be a great enhancement.

> I object. Please never do it.
> From Reddit:
> [...]

This argument sounds specific to usage on reddit, and less about a powerful keyboard app.  I appreciate the power of the app.  I use it because I prefer Dvorak layout.  Hacker's keyboard has applications far beyond a glorified usenet site.  💥

**Comment by GigabyteProductions** [comment]
Like @LinuxRunsTheWorld, I switch keyboards to send emojis, but I find it annoying. I would love if there was an option to add them (or other Unicode characters) to Hacker's Keyboard, instead

**Comment by TPS** [comment]
Duplicates #376

**Comment by GigabyteProductions** [comment]
Sorry, I happened to have seen this one on the front of the issues page. Please close this duplicate

**Comment by T0MuX** [comment]
Hi, 6 years after my post I just want you to know I found my entire satisfaction with Unexpected Keyboard. It combines, in a really great way, the "hacker part" (just have the same keyboard like on a computer, and even more btw), and the "kikoolol part" which is simply a tiny button letting you send emojis. This keyboard app is fast, easy and efficient. No ad. Open Source. Available on F-Droid. Customizable. Maintained 👀
This is just pure happiness guys.

Have a nice day :)

**Comment by User-Jeff55** [comment]
> Hi, 6 years after my post I just want you to know I found my entire satisfaction with Unexpected Keyboard. It combines, in a really great way, the "hacker part" (just have the same keyboard like on a computer, and even more btw), and the "kikoolol part" which is simply a tiny button letting you send emojis. This keyboard app is fast, easy and efficient. No ad. Open Source. Available on F-Droid. Customizable. Maintained 👀 This is just pure happiness guys.
> 
> Have a nice day :)

What is the keyboard app called?
